libwallet/src/slate.rs

```diff
@@ -525,12 +529,26 @@ impl Slate {
 			}
 		};
 
-		let blind_offset = keychain.blind_sum(
+		if self.is_compact() {
+			let total_offset = keychain.blind_sum(
+				&BlindSum::new()
+					.add_blinding_factor(self.offset.clone())
+					.add_blinding_factor(my_offset.clone()),
+			)?;
+			self.offset = total_offset;
```

Comment:
is there no more compact way to express self.offset += my_offset ?

Reply:
No in the current code, but yes it's possible to implement operator traits to support this syntax.  Again this is up in old portions of the grin Keychain code so a revision to the library could be done at a later date.